providers/aws/services/vpc/vpc_endpoint_connections_trust_boundaries/vpc_endpoint_connections_trust_boundaries_test.py

```diff
@@ -0,0 +1,201 @@
+import json
+from unittest import mock
+
+from boto3 import client
+from moto import mock_ec2
+
+AWS_REGION = "us-east-1"
+ACCOUNT_ID = "123456789012"
+
+
+class Test_vpc_endpoint_connections_trust_boundaries:
+    @mock_ec2
+    def test_vpc_no_endpoints(self):
+        from providers.aws.lib.audit_info.audit_info import current_audit_info
+        from providers.aws.services.vpc.vpc_service import VPC
+
+        current_audit_info.audited_partition = "aws"
+
+        with mock.patch(
+            "providers.aws.services.vpc.vpc_endpoint_connections_trust_boundaries.vpc_endpoint_connections_trust_boundaries.vpc_client",
+            new=VPC(current_audit_info),
+        ):
+            # Test Check
+            from providers.aws.services.vpc.vpc_endpoint_connections_trust_boundaries.vpc_endpoint_connections_trust_boundaries import (
+                vpc_endpoint_connections_trust_boundaries,
+            )
+
+            check = vpc_endpoint_connections_trust_boundaries()
+            result = check.execute()
+
+            assert len(result) == 0
+
+    @mock_ec2
+    def test_vpc_endpoint_with_full_access(self):
+        # Create VPC Mocked Resources
+        ec2_client = client("ec2", region_name=AWS_REGION)
+
+        vpc = ec2_client.create_vpc(CidrBlock="10.0.0.0/16")["Vpc"]
+
+        route_table = ec2_client.create_route_table(VpcId=vpc["VpcId"])["RouteTable"]
+        vpc_endpoint = ec2_client.create_vpc_endpoint(
+            VpcId=vpc["VpcId"],
+            ServiceName="com.amazonaws.us-east-1.s3",
+            RouteTableIds=[route_table["RouteTableId"]],
+            VpcEndpointType="Gateway",
+            PolicyDocument=json.dumps(
+                {
+                    "Statement": [
+                        {
+                            "Action": "*",
+                            "Effect": "Allow",
+                            "Principal": "*",
+                            "Resource": "*",
+                        }
+                    ]
+                }
+            ),
+        )
+        from providers.aws.lib.audit_info.audit_info import current_audit_info
+        from providers.aws.services.vpc.vpc_service import VPC
+
+        current_audit_info.audited_partition = "aws"
+
+        with mock.patch(
+            "providers.aws.services.vpc.vpc_endpoint_connections_trust_boundaries.vpc_endpoint_connections_trust_boundaries.vpc_client",
+            new=VPC(current_audit_info),
+        ):
+            # Test Check
+            from providers.aws.services.vpc.vpc_endpoint_connections_trust_boundaries.vpc_endpoint_connections_trust_boundaries import (
+                vpc_endpoint_connections_trust_boundaries,
+            )
+
+            check = vpc_endpoint_connections_trust_boundaries()
+            result = check.execute()
+
+            assert result[0].status == "FAIL"
```

Comment:
Please, check the `len(result)` and also the `region` value.

Reply:
Done